sway-core/src/semantic_analysis/ast_node/expression/typed_expression/if_expression.rs

```diff
@@ -29,19 +30,21 @@ pub(crate) fn instantiate_if_expression(
         } else {
             type_engine.insert(engines, TypeInfo::Tuple(vec![]))
         };
-        append!(
-            type_engine.unify_with_self(
-                engines,
-                then.return_type,
-                ty_to_check,
-                ctx.self_type(),
-                &then.span,
-                "`then` branch must return expected type.",
-                None
-            ),
-            warnings,
-            errors
+        let (warnings, errors) = type_engine.unify_with_self(
```

Comment:
Yeah I had an earlier version that included those, but since it isn't absolutely necessary and the change is already huge, I decided against refactoring those.

Should probably follow up with that once this is merged.